povm_toolbox/post_processor/median_of_means.py

```diff
@@ -0,0 +1,196 @@
+else:
+    from typing import override  # pragma: no cover
+
+
+import numpy as np
+from numpy.random import Generator, default_rng
+from qiskit.quantum_info import SparsePauliOp
+
+from povm_toolbox.post_processor.povm_post_processor import POVMPostProcessor
+from povm_toolbox.quantum_info.base import BaseDual
+from povm_toolbox.sampler import POVMPubResult
+
+
+class MedianOfMeans(POVMPostProcessor):
+    """A POVM result post-processor which uses a 'median of means' estimator.
+
+    This post-processor implementation provides a straight-forward interface for computing the
+    expectation values (and standard deviations) of any Pauli-based observable. It is initialized
+    with a :class:`.POVMPubResult` as shown below:
+
+    >>> from povm_toolbox.library import ClassicalShadows
+    >>> from povm_toolbox.sampler import POVMSampler
+    >>> from povm_toolbox.post_processor import MedianOfMeans
+    >>> from qiskit.circuit import QuantumCircuit
+    >>> from qiskit.primitives import StatevectorSampler
+    >>> from qiskit.quantum_info import SparsePauliOp
+    >>> circ = QuantumCircuit(2)
+    >>> _ = circ.h(0)
+    >>> _ = circ.cx(0, 1)
+    >>> povm = ClassicalShadows(2, seed=42)
+    >>> sampler = StatevectorSampler(seed=42)
+    >>> povm_sampler = POVMSampler(sampler)
+    >>> job = povm_sampler.run([circ], povm=povm, shots=16)
+    >>> result = job.result()
+    >>> post_processor = MedianOfMeans(result[0], num_batches=4, seed=42)
+    >>> post_processor.get_expectation_value(SparsePauliOp("ZI"))  # doctest: +FLOAT_CMP
+    (-0.75, 2.9154759474226504)
+
+    Additionally, this post-processor also supports the customization of the Dual frame in which the
+    decomposition weights of the provided observable are obtained. Check out
+    `this how-to guide <../how_tos/dual_optimizer.ipynb>`_ for more details on how to do this.
+    """
+
+    def __init__(
+        self,
+        povm_sample: POVMPubResult,
+        dual: BaseDual | None = None,
+        *,
+        num_batches: int | None = None,
+        delta_confidence: float | None = None,
+        seed: int | Generator | None = None,
+    ) -> None:
+        r"""Initialize the median-of-means post-processor.
+
+        Given ``num_shots=num_batches*batch_size`` samples, we partition the
+        samples into ``num_batches`` batches. We compute the mean of each batch,
+        :math:`\hat{o}_j`, and then output the median of the means, :math:`\hat{o}
+        =\mathrm{median}\{\hat{o}_1, ..., \hat{o}_{\mathrm{num\_batches}}\}`.
+        It can be shown that
+
+        .. math::
+            \lvert \mathrm{Tr}[\mathcal{O} \rho] - \hat{o} \rvert \leq \epsilon
+            \quad \textrm{with probability at least } 1-\delta \, ,
+
+        where :math:`\delta = 2 \exp{(-\mathrm{num\_batches}/2)}` and :math:`\epsilon
+        = \sqrt{\frac{34}{\mathrm{batch\_size}} } \lVert \mathcal{O} - \frac{\mathrm{Tr}
+        [\mathcal{O}]}{2^N} \mathbb{I} \rVert_\textrm{shadow}`.
```

Comment:
Just saw this. This would be better in the class docstring. Those are more prominent on the page.
The `__init__` docstring I usually limit to just explaining the input arguments.